src/node_file.cc

```diff
@@ -2307,6 +2362,77 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void Mkstemp(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  const int argc = args.Length();
+  CHECK_GE(argc, 2);
+
+  BufferValue tmpl(isolate, args[0]);
+  CHECK_NOT_NULL(*tmpl);
+
+  const enum encoding encoding = ParseEncoding(isolate, args[1], UTF8);
+
+  FSReqBase* req_wrap_async = GetReqWrap(args, 2);
+  if (req_wrap_async != nullptr) {  // mkstemp(tmpl, encoding, req)
+    AsyncCall(env,
+              req_wrap_async,
+              args,
+              "mkstemp",
+              encoding,
+              AfterMkstemp,
+              uv_fs_mkstemp,
+              *tmpl);
+  } else {  // mkstemp(tmpl, encoding, undefined, ctx)
+    CHECK_EQ(argc, 4);
+    FSReqWrapSync req_wrap_sync;
+    FS_SYNC_TRACE_BEGIN(mkstemp);
+    SyncCall(env, args[3], &req_wrap_sync, "mkstemp", uv_fs_mkstemp, *tmpl);
+    FS_SYNC_TRACE_END(mkstemp);
+    const char* path = req_wrap_sync.req.path;
+    int fd = req_wrap_sync.req.result;
+
+    Local<Value> error;
+    MaybeLocal<Value> rc = StringBytes::Encode(isolate, path, encoding, &error);
+    if (rc.IsEmpty()) {
+      Local<Object> ctx = args[3].As<Object>();
+      ctx->Set(env->context(), env->error_string(), error).Check();
+      return;
+    }
+
+    Local<Object> result = Object::New(isolate);
+    result->Set(env->context(), env->path_string(), rc.ToLocalChecked())
+        .Check();
+    result->Set(env->context(), env->fd_string(), Integer::New(isolate, fd))
+        .Check();
+    args.GetReturnValue().Set(result);
```

Comment:
Similar to https://github.com/nodejs/node/pull/33229#discussion_r427089014, I’d probably recommend passing these values as an array. Alternatively, it would be nice to not use `.Check()` here and elsewhere, since these `Set()` calls can fail.

Reply:
I pushed a commit

Reply:
Yes, exactly, although I’m guessing you want to update these lines here (where this comment points to) as well? :)

Reply:
fixed :)